### PR TITLE
Setup Stripe Coupon Workflow and Minor Refactors

### PR DESCRIPTION
**Pull request by mahadzaryab1**
### Notion ticket link

[Backend: Coupon endpoints for supporting coupon usage](https://www.notion.so/uwblueprintexecs/Backend-Coupon-endpoints-for-supporting-coupon-usage-190b91aa08cf4c27b114c97a4fbabcd7)
[Move Stripe instance to utils & add JSON response](https://www.notion.so/uwblueprintexecs/Move-Stripe-instance-to-utils-add-JSON-response-5769b2512a2c4a0190f1a8d6b96c7653)
[Stripe: Obtain price id as part of body](https://www.notion.so/uwblueprintexecs/Stripe-Obtain-price-id-as-part-of-body-4acce215e68b43788ebca1141159ad40)

### Implementation description
- Added support to Stripe checkout to enter in promo codes
- Refactored `CheckoutButton` to take the `priceID` and `quantity` as props - this will make the checkout button highly reusable 
- Refactored the checkout session endpoint to take in the `priceID` and `quantity` in the request body to allow for dynamic checkouts

### Steps to test

1. Head over to http://localhost:3000/checkout and follow the checkout workflow 
2. I've created a test coupon `TEST50` for 50% off that you can enter to test out the coupon workflow!

### Checklist

-   [x] My PR name is descriptive and in imperative tense
-   [x] I have run the linter
-   [x] I have requested a review from the PL, and/or other devs who have background knowledge on this PR or who will be building on top of this PR

### CODE REVIEW

File: pages/api/checkout/[sessionId].ts [hunk 1]
```diff
@@ -1,3 +1,4 @@
+import { ResponseUtil } from "@utils/responseUtil";
 import { NextApiRequest, NextApiResponse } from "next";
 import Stripe from "stripe";
```
Comment (feng-eric) [inline pages/api/checkout/[sessionId].ts]: Should we move Stripe to `services`?

File: pages/checkout.tsx [hunk 2]
```diff
@@ -1,5 +1,11 @@
 import { CheckoutButton } from "@components/CheckoutButton";
 
+// test price id corresponding to a product in stripe
+const testPriceId = "price_1J1GuzL97YpjuvTOePyVbsRh";
```
Comment (feng-eric) [inline pages/checkout.tsx:4]: Will this be removed in the future?

File: pages/api/checkout/[sessionId].ts (hunk 1 above)
Comment (feng-eric) [reply pages/api/checkout/[sessionId].ts]: Should it be 
```suggestion
import stripe from "@services/stripe";
```

**Review by feng-eric — COMMENTED** [review]
Had a few comments but LGTM. Were you able to test this?  Let's demo this on the meeting on Thursday

File: pages/api/checkout/[sessionId].ts (hunk 1 above)
Comment (mahadzaryab1) [reply pages/api/checkout/[sessionId].ts]: good catch - i forgot to change it in this file

File: pages/checkout.tsx (hunk 2 above)
Comment (mahadzaryab1) [reply pages/checkout.tsx:4]: this is just a test price ID but when we actually implement the page then we can likely add a field in the database to hold the priceID for the class

**Comment by mahadzaryab1** [comment]
> Had a few comments but LGTM. Were you able to test this? Let's demo this on the meeting on Thursday

Yeap! I tested it and the instructions for testing are in the PR description. I'll be ready for the demo on Thursday.

**Review by feng-eric — APPROVED** [review]
LGTM! Great work

**Pull request merged**